test/integration/timeout.integration.test.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * © Copyright IBM Corporation 2021. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const assert = require('assert');
+const sinon = require('sinon');
+
+const { CloudantV1 } = require('../../index.ts');
+const { CouchdbSessionAuthenticator } = require('../../index.ts');
+const {
+  BasicAuthenticator,
+  IamAuthenticator,
+  NoAuthAuthenticator,
+} = require('ibm-cloud-sdk-core');
+
+const TESTCASE_TIMEOUT = 10000; // (10s)
+
+const DEFAULT_TIMEOUT = 150000; // (2.5m=150s)
+const CUSTOM_TIMEOUT = 30000; // (30s)
+
+describe('Default timeout config tests', () => {
+  jest.setTimeout(TESTCASE_TIMEOUT);
+
+  function assertBaseTimeoutOptions(myService, expTimeoutValue) {
+    assert.ok(myService.baseOptions.timeout);
+    assert.equal(myService.baseOptions.timeout, expTimeoutValue);
+  }
+
+  function assertAuthTokenTimeoutOptions(myService, expTimeoutValue) {
+    const auth = myService.getAuthenticator();
+    assert.ok(auth.tokenOptions.timeout);
+    assert.equal(auth.tokenOptions.timeout, expTimeoutValue);
+  }
+
+  // Every test case tests an authenticator.
+  // 1. case: its default timeout value checked,
+  // 2.case: a custom overwrite possibility is checked.
+  const testCases = [
+    // Default
+    {
+      options: {
+        authenticator: {},
+      },
+      expTimeout: DEFAULT_TIMEOUT,
+    },
+    // Overwrite
+    {
+      options: {
+        authenticator: {},
+        timeout: CUSTOM_TIMEOUT,
+      },
+      expTimeout: CUSTOM_TIMEOUT,
+    },
+  ];
+
+  it('CloudantV1 - BasicAuth', () => {
+    const basicAuth = new BasicAuthenticator({
+      username: 'user',
+      password: 'pwd',
+    });
+
+    for (const tc of testCases) {
```

Comment:
I would prefer the current structure.

Reply:
As you wish! 🙂 

This was just an improvement idea, I always try to find the golden mean between [DRY and DUMP principles](https://testing.googleblog.com/2019/12/testing-on-toilet-tests-too-dry-make.html) when I write tests, and this for loop IMO is not so DUMP, but I can live with it, as soon as it will not waste the time to read it at maintenance.

Reply:
In my opinion it is more important to highlight the structure of the timeout test cases.
There 4 Authenticator and each tested with default and custom timeout config. That's all that a quick test review should know when there is some problem with 1 of the methods.
This is a structure that a new authenticator should also follow.

I think the current structure is easily readable.